src/amo/components/SearchSort/index.js

```diff
@@ -0,0 +1,79 @@
+import classNames from 'classnames';
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+
+import SearchSortLink from 'amo/components/SearchSortLink';
+import translate from 'core/i18n/translate';
+
+import './SearchSort.scss';
+
+
+export class SearchSortBase extends React.Component {
+  static propTypes = {
+    _searchToggle: PropTypes.object.isRequired,
+    filters: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+    pathname: PropTypes.string.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { sortVisible: false };
+  }
+
+  onClick = (event) => {
+    event.preventDefault();
+    this.toggleSort(this.props._searchToggle || this.searchToggle);
+  }
+
+  sortOptions() {
+    const { i18n } = this.props;
+
+    return [
+      { sort: 'updated', text: i18n.gettext('Recently Updated') },
+      { sort: 'relevance', text: i18n.gettext('Relevance') },
+      { sort: 'users', text: i18n.gettext('Most Users') },
+      { sort: 'rating', text: i18n.gettext('Top Rated') },
+    ];
+  }
+
+  toggleSort(searchToggle) {
+    // const searchToggle = _searchToggle || this.searchToggle;
+    // Blur the search sort toggle if we're closing it.
+    if (this.state.sortVisible) {
+      searchToggle.blur();
+    }
+    this.setState({ sortVisible: !this.state.sortVisible });
+  }
+
+  render() {
+    const { filters, i18n, pathname } = this.props;
+    const { sortVisible } = this.state;
+    const currentSort = filters.sort || 'relevance';
+
+    return (
+      <div className={classNames('SearchSort', {
+        'SearchSort--visible': sortVisible,
+      })}>
+        <a className="SearchSort-toggle" href="#SearchSortOptions"
+          ref={(ref) => { this.searchToggle = ref; }} onClick={this.onClick}>
+          {i18n.gettext('Sort')}
```

Comment:
I guess. Problem with that is I'm not sure where the icon should be LTR vs RTL and I don't feel like it's content so including it in the link feels weird. I like the idea of the Icon component (I saw it and my first crack at this used that), but in usage it feels odd, especially as there are two icons here (one left, one right).

I guess in this one it's not too bad, I'll take a crack at it.